docs/reference/dlm/apis/get-lifecycle.asciidoc

```diff
@@ -0,0 +1,143 @@
+[[dlm-get-lifecycle]]
+=== Get the lifecycle of a data stream
+++++
+<titleabbrev>Get Data Stream Lifecycle</titleabbrev>
+++++
+
+experimental::[]
+
+Gets the lifecycle of a set of data streams.
+
+[[dlm-get-lifecycle-request]]
+==== {api-request-title}
+
+`GET _data_stream/_lifecycle`
+`GET _data_stream/<data-stream>/_lifecycle`
+
+[[dlm-get-lifecycle-prereqs]]
+==== {api-prereq-title}
+
+// TODO Update security
+// * If the {es} {security-features} are enabled, you must have the <role>
+// cluster privilege to use this API. For more information, see
+// <<security-privileges>>.
+
+[[dlm-get-lifecycle-desc]]
+==== {api-description-title}
+
+Gets the lifecycle of the specified data streams. If a data stream does not have a lifecycle defined then it will
+be ommited from the list.
```

Comment:
If only one data stream is specified we should be returning empty for lifecycle right? e.g. `{}`

Reply:
Hm, I wasn't aware of this, right now it will return an empty list not an empty object. I am not a fan of changing the structure of a response based on if you are asking one or more. I prefer to always return a list honestly, then the user can always parse it in a predictable way. What do you think?

Reply:
Oh, sorry for not being verbose enough. 

I was referring to keeping the array but including the requested **data stream name** in the response (as opposed to omitting it because it doesn't have a lifecycle)

e.g.
```
{
  "data_streams": [
      {
        "name": "my-data-stream-1"
      }
   ]
}
```

I wonder if omitting it would make clients a bit unhappy as it could mean that ... the data stream doesn't exist? Or just that it doesn't have a lifecycle?
Do we return 404 if just one data stream is requested and it doesn't exist?

Reply:
Ah, I see below we 404 for data streams that don't exist, so ignore the above :) 

I think we should just document that if multiple data streams are requested we 404 the entire request if one is missing

Reply:
Very good point, I will do that.